experimental/generation/generator/packages/library/src/dialogGenerator.ts

```diff
@@ -781,34 +796,40 @@ export async function generate(
 
         await processTemplates(schema, templateDirs, allLocales, outPath, scope, force, feedback)
 
-        // Expand schema expressions
+        // Expand all remaining schema expressions
         let expanded = expandSchema(schema.schema, scope, '', false, true, feedback)
 
-        // Write final schema
-        let body = stringify(expanded, (key: any, val: any) => (key === '$templates' || key === '$requires' || key === '$templateDirs' || key === '$examples') ? undefined : val)
-        await generateFile(ppath.join(outPath, `${prefix}.json`), body, force, feedback)
+        if (!error) {
+            // Write final schema
+            let body = stringify(expanded, (key: any, val: any) => (key === '$templates' || key === '$requires' || key === '$templateDirs' || key === '$examples') ? undefined : val)
+            await generateFile(ppath.join(outPath, `${prefix}.json`), body, force, feedback)
 
-        // Merge together all dialog files
-        if (singleton) {
-            if (!merge) {
-                feedback(FeedbackType.info, 'Combining into singleton .dialog')
-                await generateSingleton(scope.prefix, outPath, outDir, feedback)
-            } else {
-                await generateSingleton(scope.prefix, outPath, outPathSingle, feedback)
+            // Merge together all dialog files
+            if (singleton) {
```

Comment:
Nit: Since generateSingleton and merge calls differ only by outDir vs outPathSingle, you could set a workingOutDir variable with the logic and then make the calls without multiple if statements.